test/extensions/filters/http/lua/lua_filter_test.cc

```diff
@@ -2392,6 +2392,31 @@ TEST_F(LuaHttpFilterTest, LuaFilterSetResponseBufferChunked) {
   EXPECT_EQ(Http::FilterDataStatus::Continue, filter_->encodeData(response_body, true));
 }
 
+TEST_F(LuaHttpFilterTest, LuaBodyBufferSetBytesWithHex) {
```

Comment:
We also need a use case to test the string contains zero.

Reply:
>  contains zero
Does it means empty string?

Reply:
I may not describe it clearly. I mean a string containing embedded zeros ('\0'), not an empty string.  @Patrick0308